CONTRIBUTING.md

```diff
@@ -113,6 +113,40 @@ the repository maintainers will integrate it, prepare changelogs, and
 ensure it gets released in one of upcoming Weekly releases.
 There is no additional action required from pull request authors at this point.
 
+### Pull-Requests management
+
+The project is using some labels to mark the status and the content of the pull-requests.
+The complete list of labels can be found at https://github.com/jenkinsci/jenkins/labels.
+Out of those labels, here are a definition of labels the maintainers are using to manage the status of the pull-requests:
+
+- `needs-docs` marks a pull-requests lacking documentation in the code or on jenkins.io ;
+such pull-requests won't be merged until the comments are addressed
+- `needs-fix` marks a pull-requests with reviews requesting some code change which are not addressed yet ;
+such pull-requests won't be merged until the code has been fixed and the tests are passing
+- `needs-justification` marks pull-requests on which maintainers are debating the motivation for the proposed changed
```

Comment:
I suggest making this more action-oriented for new users by explaining to them exactly what kind of justification is desired in order to result in the most efficient review of their PR. Here is some text I have written in the past; feel free to borrow from this and/or summarize it:

> Provide a clear description of the high-level effort with which this pull request is associated. Recall that while anyone in the project can see this pull request, not everyone necessarily has the same context as you. If applicable, link to design documents or high-level tracking epics here. Also provide a clear description of the high-level problem you are trying to solve. The problem statement should be written in terms of a specific symptom that affects users or the project. The problem statement should not be written in terms of the solution. If possible, include a minimal reproducible example (MRE) with steps to reproduce, expected results, and actual results. If the cause of the problem is not obvious, provide a root-cause analysis (RCA), ideally using the Five Whys iterative interrogative technique or some other form of analytical reasoning. Finally, provide a clear description of the high-level solution you have chosen and describe, explicitly, how it addresses the high-level problem. If there were other possible solutions that you considered and rejected, mention those along with the corresponding reasoning.